main/src/cgeo/geocaching/filters/gui/GeocacheFilterActivity.java

```diff
@@ -383,6 +408,18 @@ public static void selectFilter(@NonNull final Activity context, final GeocacheF
         context.startActivityForResult(intent, REQUEST_SELECT_FILTER);
     }
 
+    public void selectNestedFilter(final LogicalFilterViewHolder holder) {
+        final GeocacheFilterContext nestedFilterContext = new GeocacheFilterContext(TRANSIENT);
+        nestedFilterContext.set(GeocacheFilter.create(null, true,
+                includeInconclusiveFilterCheckbox.isChecked(), holder.createFilterFromView()));
+
+        final Intent intent = new Intent(this, GeocacheFilterActivity.class);
+        intent.putExtra(EXTRA_FILTER_CONTEXT, nestedFilterContext);
```

Comment:
I believe you don't need a GeocacheFilterContext at all. You could simply pass the filter's config as a string:

`holder.createFilterFromView().toConfig()`

From such a config, you can always recreate a GeocacheFilter object using static method `GeocacheFilter.createFromConfig`

Reply:
A filter context gets created anyway inside the `GeocacheFilterActivity`. So why not use it to pass the current filter...?

![grafik](https://user-images.githubusercontent.com/64581222/168091490-a7f66bf3-518a-4a5f-8803-e850b1ff84bf.png)

Reply:
Hm I guess you're right. Then just leave it as is